packages/react-components/lib/task-summary/create-task.tsx

```diff
@@ -0,0 +1,460 @@
+          </Button>
+        </Grid>
+      )}
+    </Grid>
+  );
+}
+
+interface DeliveryTaskFormProps {
+  value: DeliveryTaskDescription;
+  onChange(deliveryTaskDescription: DeliveryTaskDescription): void;
+}
+
+function DeliveryTaskForm({ value, onChange }: DeliveryTaskFormProps) {
+  const theme = useTheme();
+
+  return (
+    <>
+      <Grid container wrap="nowrap">
+        <Grid style={{ flex: '1 1 60%' }}>
+          <TextField
+            id="pickup-location"
+            label="Pickup Location"
+            fullWidth
+            margin="normal"
+            value={value.pickupPlaceName}
+            onChange={(ev) =>
+              onChange({
+                ...value,
+                pickupPlaceName: ev.target.value,
+              })
+            }
+          />
+        </Grid>
+        <Grid
+          style={{
+            flex: '1 1 40%',
+            marginLeft: theme.spacing(2),
+            marginRight: theme.spacing(2),
+          }}
+        >
+          <TextField
+            id="dispenser"
+            label="Dispenser"
+            fullWidth
+            margin="normal"
+            value={value.pickupDispenser}
+            onChange={(ev) => {
+              onChange({
+                ...value,
+                pickupDispenser: ev.target.value,
+              });
+            }}
+          />
+        </Grid>
+      </Grid>
+      <Grid container wrap="nowrap">
+        <Grid style={{ flex: '1 1 60%' }}>
+          <TextField
+            id="dropoff-location"
+            label="Dropoff Location"
+            fullWidth
+            margin="normal"
+            value={value.dropoffPlaceName}
+            onChange={(ev) =>
+              onChange({
+                ...value,
+                dropoffPlaceName: ev.target.value,
+              })
+            }
+          />
+        </Grid>
+        <Grid
+          style={{
+            flex: '1 1 40%',
+            marginLeft: theme.spacing(2),
+            marginRight: theme.spacing(2),
+          }}
+        >
+          <TextField
+            id="ingestor"
+            label="Ingestor"
+            fullWidth
+            margin="normal"
+            value={value.dropoffIngestor}
+            onChange={(ev) => {
+              onChange({
+                ...value,
+                dropoffIngestor: ev.target.value,
+              });
+            }}
+          />
+        </Grid>
+      </Grid>
+    </>
+  );
+}
+
+interface LoopTaskFormProps {
+  value: LoopTaskDescription;
+  onChange(loopTaskDescription: LoopTaskDescription): void;
+}
+
+function LoopTaskForm({ value, onChange }: LoopTaskFormProps) {
+  const theme = useTheme();
+
+  return (
+    <>
+      <TextField
+        id="start-location"
+        label="Start Location"
+        fullWidth
+        margin="normal"
+        value={value.startName}
+        onChange={(ev) =>
+          onChange({
+            ...value,
+            startName: ev.target.value,
+          })
+        }
+      />
```

Comment:
Maybe I'm missing something, but shouldn't we provide available places for the robot to go?

Same with the delivery request and the dispenser-ingestor

Reply:
I left that feature as todo for this initial implementation. Particularly because there is no formal way to get other infos like cleaning zones, (and future task types like patrol zones, charging stations etc).